crates/domain/src/repositories/task_repository.rs

```diff
@@ -0,0 +1,125 @@
+use std::sync::Arc;
+use axum::async_trait;
+use chrono::Utc;
+use sqlx::{Pool, Postgres, query, Row};
+use sqlx::postgres::PgRow;
+use crate::models::task::Task;
+
+pub type DynTaskRepository = Arc<dyn TaskRepository + Send + Sync>;
+
+#[async_trait]
+pub trait TaskRepository {
+    async fn get_tasks(&self) -> anyhow::Result<Vec<Task>>;
+    async fn create_task(&self, task: Task) -> anyhow::Result<Task>;
+}
+
+#[derive(Clone)]
+pub struct DomainTaskRepository {
+    pool: Pool<Postgres>,
+}
+
+impl DomainTaskRepository {
+    pub fn new(pool: Pool<Postgres>) -> Self {
+        Self { pool }
+    }
+}
+
+#[async_trait]
+impl TaskRepository for DomainTaskRepository {
+    async fn get_tasks(&self) -> anyhow::Result<Vec<Task>> {
+        let tasks: Vec<Task> = query(
+            "
+                SELECT
+                    tasks.id,
+                    tasks.user_id,
+                    tasks.name,
+                    tasks.description,
+                    tasks.status,
+                    tasks.created_at,
+                    tasks.updated_at,
+                    COALESCE(array_agg(tags.name), ARRAY[]::VARCHAR[]) AS tags
+                FROM tasks
+                LEFT JOIN tasks_tags ON tasks.id = tasks_tags.task_id
+                LEFT JOIN tags ON tasks_tags.tag_id = tags.id
+                GROUP BY tasks.id
+                ORDER BY tasks.id
+            ")
+                .map(|row: PgRow| Task {
+                    id: row.get("id"),
+                    user_id: row.get("user_id"),
+                    name: row.get("name"),
+                    description: row.get("description"),
+                    status: row.get("status"),
+                    tags: row.get("tags"),
+                    created_at: row.get("created_at"),
+                    updated_at: row.get("updated_at"),
+                })
+                .fetch_all(&self.pool)
+                .await?;
+        Ok(tasks)
+    }
+
+    async fn create_task(&self, task: Task) -> anyhow::Result<Task> {
+        let now = Utc::now().naive_utc();
```

Comment:
Why do we want to remove timezone information?

Reply:
Seems like `Utc::now().naive_utc()` is commonly used when you need a datetime for purposes like storing in a database, logging, or comparing without explicitly requiring timezone handling. (So the timezone is always recorded in the same way).

Reply:
https://docs.rs/chrono/latest/chrono/naive/struct.NaiveDateTime.html

Yes, but we may want to keep the timezone information and the column time is also a `TIMESTAMPTZ`, which is with timezone information already.